cpp/iedriver/IECommandExecutor.cpp

```diff
@@ -330,38 +333,55 @@ LRESULT IECommandExecutor::OnAfterNewWindow(UINT uMsg,
         }
 
         if (diff.size() == 0) {
+          MSG msg;
+          if (loopCount >= 1 && !isProcessWM &&::PeekMessage(&msg, NULL, WM_USER, WM_USER, PM_REMOVE)) {
+              ::TranslateMessage(&msg);
+              ::DispatchMessage(&msg);
+              isProcessWM = true;
+              LOG(TRACE) << "process WM_USER";
+              ::Sleep(500);
+          }
           ::Sleep(500);
         }
+        loopCount++;
       }
 
       if (diff.size() == 0) {
         LOG(WARN) << "No new window handle found after attempt to open";
       } else {
-        HWND new_window_window = diff[0];
-
-        DWORD process_id;
-        ::GetWindowThreadProcessId(new_window_window, &process_id);
-        clock_t end = clock() + (DEFAULT_BROWSER_REATTACH_TIMEOUT_IN_MILLISECONDS / 1000 * CLOCKS_PER_SEC);
-        bool is_ready = this->factory_->IsBrowserProcessInitialized(process_id);
-        while (!is_ready && clock() < end) {
-          ::Sleep(100);
-          is_ready = this->factory_->IsBrowserProcessInitialized(process_id);
-        }
+        for (int i = diff.size() - 1; i >= 0; i--) {
+          HWND new_window_window = diff[i];
+
+          DWORD process_id = 0;
+          ::GetWindowThreadProcessId(new_window_window, &process_id);
+          if (process_id) {
+            clock_t end = clock() + (DEFAULT_BROWSER_REATTACH_TIMEOUT_IN_MILLISECONDS / 1000 * CLOCKS_PER_SEC);
+            bool is_ready = this->factory_->IsBrowserProcessInitialized(process_id);
+            while (!is_ready && clock() < end) {
+              ::Sleep(100);
+              is_ready = this->factory_->IsBrowserProcessInitialized(process_id);
+            }
+
+            ProcessWindowInfo info;
+            info.dwProcessId = process_id;
+            info.hwndBrowser = new_window_window;
+            info.pBrowser = NULL;
+            std::string error_message = "";
+            bool attachFlag = this->factory_->AttachToBrowser(&info, &error_message);
```

Comment:
```suggestion
            bool attach_flag = this->factory_->AttachToBrowser(&info, &error_message);
```

Reply:
thank you for your suggestions.
I've renamed variables.